src/util/get-github-client-config.ts

```diff
@@ -22,17 +22,52 @@ interface GitHubClientConfig extends GitHubConfig {
 	gitHubClientSecret: string;
 }
 
-export async function getGitHubApiUrl(jiraHost: string, gitHubAppId: number) {
-	const gitHubClientConfig = await getGitHubClientConfigFromAppId(gitHubAppId, jiraHost);
+export async function getGitHubApiUrl(jiraHost: string, gitHubAppId: number, logger: Logger) {
+	const gitHubClientConfig = await getGitHubClientConfigFromAppId(gitHubAppId, logger, jiraHost);
 	return await booleanFlag(BooleanFlags.GHE_SERVER, GHE_SERVER_GLOBAL, jiraHost) && gitHubClientConfig
 		? `${gitHubClientConfig.apiUrl}`
 		: GITHUB_CLOUD_API_BASEURL;
 }
 
-export const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined, jiraHost?: string): Promise<GitHubClientConfig> => {
+/**
+ * Populates proxyBaseUrl and returns the object itself (for convenient chaining)
+ * @param config - object to setup proxy in
+ * @param logger
+ * @param gitHubBaseUrl - for cloud, either undefined or "github.com"'s base url
+ * @param jiraHost - jiraHost from context
+ */
+async function setupProxyConfig<T extends GitHubConfig>(config: T, logger: Logger, jiraHost: string, gitHubBaseUrl?: string): Promise<T> {
+	if (!await booleanFlag(BooleanFlags.USE_OUTBOUND_PROXY_SKIPLIST, false, jiraHost)) {
+		return config;
+	}
+
+	if (gitHubBaseUrl && gitHubBaseUrl != GITHUB_CLOUD_BASEURL) {
+		const skipList = await stringFlag(StringFlags.OUTBOUND_PROXY_SKIPLIST, "", jiraHost);
+		let skipOutboundProxy;
+		try {
+			skipOutboundProxy = skipList
+				.split(',')
+				.filter(hostname => !!hostname)
+				.map(hostname => hostname.trim())
+				.map(hostname => hostname.indexOf("://") >= 0 ? hostname : "http://" + hostname)
+				.map(hostname => new URL(hostname).host.toLowerCase())
+				.indexOf(new URL(gitHubBaseUrl).host.trim().toLowerCase()) >= 0;
+		} catch (err) {
+			logger.error({ err }, "Cannot evaluate skiplist because of a error, opting for outboundproxy for good");
+			skipOutboundProxy = false;
+		}
+		if (skipOutboundProxy) {
+			return config;
+		}
+	}
+	config.proxyBaseUrl = envVars.PROXY;
+	return config;
+}
+
+export const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined, logger: Logger, jiraHost: string): Promise<GitHubClientConfig> => {
 	const gitHubServerApp = gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId);
 	if (gitHubServerApp) {
-		return	{
+		return await setupProxyConfig({
```

Comment:
I am not a fan of this pattern as you're making the original config much harder to read and follow, and can also produce side effects.  Instead I'd recommend you do 
```
return {
   serverId: gitHubServerApp.id,
   ...
   proxyBaseUrl: this.setupProxyConfig(logger, jiraHost, gitHubServerApp.gitHubBaseUrl)
   };
```

Also, as a standard, we've been moving logger to the last argument in a function list as the other information is more important.